src/shared/Table/ResponsiveTable.tsx

```diff
@@ -190,7 +201,19 @@ export const ResponsiveTable = <TRow, TCol>({
       variant={variant}
     >
       <Thead>
-        <Tr>{header}</Tr>
+        <Tr>
+          {onBulkSelect !== undefined && (
```

Comment:
So if I pass `onBulkSelect` but on `onCheck` I get the checkbox on the header but not on the rows.

On the render function, we must check for the correctness of compound props, and `throw` an exception if they are not used as they should so to warn the developer about the mistake.

Also, name consistency please. If we used `onCheck`, then it should be `onBulkCheck`. We are using `select` for the action of highlighting a row, so this can be mistaken as a way to highlight all rows at once (which can't be done).